### PR TITLE
Bump up chainer version

### PR DESCRIPTION
**Pull request by ir5**
to 4381df84878e08f55f0cba31759fc7628120c449

This will enable to use hook functions in ChainerX memory pool.

### CODE REVIEW

File: scripts/large_models/chainer_chain.py [hunk 1]
```diff
@@ -10,7 +10,7 @@
 from chainer.links.connection.convolution_2d import Convolution2D
 from chainer.links.connection.linear import Linear
 from chainer.functions.noise.dropout import dropout
-from chainer.functions.pooling.max_pooling_2d import max_pooling_2d
+from chainer.functions.pooling.max_pooling_nd import max_pooling_2d
```
Comment (shinh) [inline scripts/large_models/chainer_chain.py]: This is fine, but cannot we use `chainer.functions import max_pooling_2d` so we can support both old chainer and new one?

Comment (ir5) [reply scripts/large_models/chainer_chain.py]: Thank you, fixed.

**Pull request merged**